pipeline/container/frontend/index.html

```diff
@@ -759,57 +972,265 @@
+            const renderedFiles = value.map((file, idx) => {
+              return <li key={idx}>{renderFile(file.file)}</li>;
+            });
+            return <ul className="flex flex-col gap-6">{renderedFiles}</ul>;
+          };
+
+          // => RunOutputFile
+          const renderFile = (file) => {
+            return (
+              // TODO: Add a tab for the file contents, and it's raw response
+              <PipelineFileResponse file={file} />
+            );
+          };
+
+          // => string
+          const renderString = (value) => {
+            return (
+              <pre class="bg-gray-700 text-white rounded p-2 overflow-auto">
+                {value}
+              </pre>
+            );
+          };
+
+          // => any[]
+          const renderArray = (value) => {
+            // If array of files
+            let isArrayOfFiles = value.find((x) => x.type == "file");
+            if (isArrayOfFiles) {
+              return renderFiles(value);
+            }
+
+            // If array of other types, print full array
+            return (
+              <ul className="flex flex-col gap-6">
+                <pre class="bg-gray-700 text-white rounded p-2 overflow-auto">
+                  {JSON.stringify(value, null, 2)}
+                </pre>
+              </ul>
+            );
+          };
+
+          // Render Object function
+          const renderObject = (value) => (
+            <ul className="flex flex-col gap-4">
+              <pre class="bg-gray-700 text-white rounded p-2 overflow-auto">
+                {JSON.stringify(value, null, 2)}
+              </pre>
+            </ul>
+          );
+
+          switch (output.type) {
+            case "file":
+              return renderFile(output.file);
+
+            case "dictionary":
+              return renderObject(output.value);
+
+            case "string":
+              return renderString(output.value);
+
+            case "array":
+              return renderArray(output.value);
+          }
+
+          // Edge cases: If not known type, continue to render special cases
+          if (isObject(output)) {
+            return renderObject(output);
+          }
+          if (isArray(output)) {
+            return renderArray(output);
+          }
+
+          return null;
         }
 
+        // GET pipeline
+        React.useEffect(() => {
+          fetch(`${host}/v4/container/pipeline`)
+            .then((res) => res.json())
+            .then((data) => {
+              setPipeline(data);
+
+              const dynamicFields = generateDynamicFieldsFromIOVariables(
+                data.input_variables
+              );
+              // const dynamicFields =
+              //   generateDynamicFieldsFromIOVariables(mockInputIOVariables);
+
+              setDynamicFields(dynamicFields);
+
+              const inputData =
+                generateInputDataArrayFromDynamicFields(dynamicFields);
+              setInputData(inputData);
```

Comment:
I guess this is not needed anymore due the useEffect, e.g. inputData will be set twice on mount. But no biggie if you want to leave it and merge